lib/hedwig_mopidy/responders/mopidy.ex

```diff
@@ -1,175 +1,249 @@
 defmodule HedwigMopidy.Responders.Mopidy do
   use Hedwig.Responder
 
-  alias Mopidy.{Library,Tracklist,Playback}
-  alias Mopidy.{Track,TlTrack,SearchResult}
+  alias Mopidy.{Library,Tracklist,Playback,Playlists,Playlist}
+  alias Mopidy.{Track,TlTrack,SearchResult,Ref}
 
-  hear ~r/^mopidy$/i, message do
-    response = "Hedwig Mopidy\n"
-    
-    response =
-      if web_url = HedwigMopidy.web_url do
-        response <> "Web URL: " <> web_url <> "\n"
-      else
-        response <> "Web URL: No web URL set\n"
-      end
+  defmodule Thumb do
+    defstruct [:user, :track, :playlist, :direction, :timestamp]
+  end
 
-    response =
-      if icecast_url = HedwigMopidy.icecast_url do
-        response <> "Icecast URL: " <> icecast_url <> "\n"
-      else
-        response
+  defmodule ThumbStore do
+    defp brain do
+      HedwigBrain.brain
+    end
+
+    defp storage do
+      brain.get_lobe("thumbs")
+    end
+
+    def store(%Thumb{} = data, name) do
+      brain.put(storage, name, data)
+      data
+    end
+
+    def retrieve(name) do
+      case brain.get(storage, canonicalize(name)) do
+        nil -> %Thumb{}
+        data -> data
       end
+    end
+
+    def all do
+      brain.all(storage)
+    end
 
-    send message, HedwigMopidy.playing_message(response)
+    def canonicalize(string) do
+      string
+      |> String.downcase
+      |> String.trim
+    end
   end
 
-  hear ~r/play artist (?<artist>.*)/i, message do
-    artist = message.matches["artist"]
+  defmodule CurrentPlaylistStore do
+    defp brain do
+      HedwigBrain.brain
+    end
 
-    response = 
-      with {:ok, %SearchResult{} = search_results} <- Library.search(%{artist: [artist]}),
-           {:ok, :success} <- Tracklist.clear,
-           {:ok, tracks} when is_list(tracks) <- Tracklist.add(search_results.tracks |> Enum.map(fn(%Track{} = track) -> track.uri end)),
-           {:ok, :success} <- Playback.play do
-        HedwigMopidy.currently_playing
-      else
-        {:error, error_message} -> error_message
-        _ ->
-          case Tracklist.get_length do
-            {:ok, 0} -> HedwigMopidy.error_message("Couldn't find any music for that artist")
-            _        -> HedwigMopidy.error_message("Couldn't play music by that artist")
-          end
+    defp storage do
+      brain.get_lobe("playlist")
+    end
+
+    def store(%Playlist{} = playlist) do
+      brain.put(storage, "current_playlist", playlist)
+      playlist
+    end
+
+    def retrieve do
+      case brain.get(storage, "current_playlist") do
+        nil -> %Playlist{}
+        playlist -> playlist
       end
+    end
+
+    def all do
+      brain.all(storage)
+    end
+  end
 
+  respond ~r/dj$/i, message do
```

Comment:
This looks like it could just go into `@usage`, no? Could define it in `@usage` and still print it here w/o any duplication